### PR TITLE
Update documentation

### LINKED ISSUE #470: Documentation updates for training
realted to #463
Description:
-------------
Go over CUQIpy documentation and make light updates to help improve the documentation (e.g. fixes of typos, equations not rendered correctly) and identify larger documentation update tasks that are necessary

Definition of done:
-------------------
- [x] Light update (as described above) is done, reviewed, and merged
- [ ] Bigger and necessary documentation update tasks are identified and corresponding issues are created, assigned and linked from this issue. (No major issues found at this stage.)
- [x] The possibility of fixing errors related to documentation that appear when CUQIpy is installed on a Windows machine is investigated
- [x] #488
- [x] Add experimental mcmc to main API page

Notes on Windows installation:
------------------------------
- @chaozg went through the CUQIpy installation process on Windows with Tania by following the instructions listed at the newest CUQIpy [documentation](https://cuqi-dtu.github.io/CUQIpy/user/getting_started.html), and everything worked fine. Just some notes:
  - After installation, it might take some time, like half a minute, for vscode to update its kernel options, so the just installed environment appears as an available python kernel
  - For working with the terminal interface in vscode, we should choose "terminal prompt" as the terminal type, as other terminals don't know conda

### PR DESCRIPTION
**Pull request by chaozg**
fix #470 

Major changes I made:
- removed the manually inserted sections of `Attributes` and `Methods` from the docstring of class. Instead, we move respective docstrings to the definition of relevant properties and methods. Then we rely on sphinx to automatically capture the docstring.
- updated math equations, mainly in distribution and its subclasses, so the equations are rendered correctly.
- added `experiment.mcmc` to the overall API page, which will appear at the bottom of https://cuqi-dtu.github.io/CUQIpy/api/index.html

### CODE REVIEW

File: cuqi/distribution/_modifiedhalfnormal.py [hunk 1]
```diff
@@ -7,14 +7,17 @@
 class ModifiedHalfNormal(Distribution):
     """
     Represents a modified half-normal (MHN) distribution, a three-parameter family of distributions generalizing the Gamma distribution.
-    The distribution is continuous with pdf 
-    f(x; alpha, beta, gamma) propto x^(alpha-1) * exp(-beta * x^2 + gamma * x)
+    The distribution is continuous with pdf
+
+    .. math::
+    
+        f(x; \\alpha, \\beta, \\gamma) \propto x^{(\\alpha-1)} * \exp(-\\beta * x^2 + \\gamma * x)
```
Comment (amal-ghamdi) [inline cuqi/distribution/_modifiedhalfnormal.py:14]: one note here, using alpha for example as a word, shows the direct link with the parameters. However, having it as a symbol is much nicer. May I suggest in the definition of parameters below that you add the symbol to make the link explicit, for example:
```
    Parameters
    ----------
    alpha : float
        The polynomial exponent parameter :math:`\\alpha` of the MHN distribution. Must be positive.
    beta : float
        The quadratic exponential parameter :math:`\\beta` of the MHN distribution. Must be positive.
    gamma : float
        The linear exponential parameter :math:`\\gamma` of the MHN distribution.
```

File: cuqi/distribution/_beta.py [hunk 2]
```diff
@@ -9,23 +9,17 @@ class Beta(Distribution):
     """
     Multivariate beta distribution of independent random variables x_i. Each is distributed according to the PDF function
 
-    f(x) = x^(alpha-1) * (1-x)^(beta-1) * Gamma(alpha+beta) / (Gamma(alpha)*Gamma(beta))
+    .. math::
 
-    where Gamma is the Gamma function.
+        f(x) = x^{(\\alpha-1)}(1-x)^{(\\beta-1)}\Gamma(\\alpha+\\beta) / (\Gamma(\\alpha)\Gamma(\\beta))
+
+    where :math:`\Gamma` is the Gamma function.
```
Comment (amal-ghamdi) [inline cuqi/distribution/_beta.py:16]: it is nice to have the formula updated with symbols. I suggest to link the class parameters with the equation to be more explicit (may be something like this)
```
    Parameters
    ------------
    alpha: float or array_like
        The shape parameter :math:`\\alpha` of the beta distribution

    beta: float or array_like
        The shape parameter :math:`\\beta` of the beta distribution
```

File: cuqi/distribution/_inverse_gamma.py [hunk 3]
```diff
@@ -8,9 +8,11 @@ class InverseGamma(Distribution):
     """
     Multivariate inverse gamma distribution of independent random variables x_i. Each is distributed according to the PDF function
 
-    f(x) = (x-location)^(-shape-1) * exp(-scale/(x-location)) / (scale^(-shape)*Gamma(shape))
+    .. math::
 
-    where shape, location and scale are the shape, location and scale of x_i, respectively. And Gamma is the Gamma function.
+        f(x) = (x-\\beta)^{(-\\alpha-1)} * \exp(-\\gamma/(x-\\beta)) / (\\gamma^{(-\\alpha)}*\Gamma(\\alpha))
+
+    where shape :math:`\\alpha`, location :math:`\\beta` and scale :math:`\\gamma` are the shape, location and scale of x_i, respectively. And Gamma is the Gamma function.
```
Comment (amal-ghamdi) [inline cuqi/distribution/_inverse_gamma.py:15]:
```suggestion
    where shape :math:`\\alpha`, location :math:`\\beta` and scale :math:`\\gamma` are the shape, location and scale of x_i, respectively. And :math:`\Gamma` is the Gamma function.
```

**Review by amal-ghamdi — CHANGES_REQUESTED** [review]
Many thanks @chaozg for this update!, It addresses the DoD nicely.

I added a few suggestions for you to address as you see fit.

File: cuqi/distribution/_modifiedhalfnormal.py (hunk 1 above)
Comment (chaozg) [reply cuqi/distribution/_modifiedhalfnormal.py:14]: Done as suggested

File: cuqi/distribution/_beta.py (hunk 2 above)
Comment (chaozg) [reply cuqi/distribution/_beta.py:16]: Done as suggested

**Comment by chaozg** [comment]
Thank you @amal-ghamdi and I just included your suggestions in the last few commits, and am now re-requesting your and @jakobsj 's review.

**Review by amal-ghamdi — APPROVED** [review]
Thank you @chaozg, LGTM

**Review by jakobsj — APPROVED** [review]
Thank you, updates look very good. (I didn't check the math updates very carefully, so I trust you have taken care when updating that!)

**Pull request merged**